File: Backend/Module.Users.Contracts/CurrentUser/Queries/GetUser/UserViewModel.cs
using Module.Companies.Contracts.Common;
using Shared.Models;

namespace Module.Users.Contracts.CurrentUser.Queries.GetUser
{
    public class UserViewModel
    {
        public int Id { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public RoleEnum Role { get; set; }
        public int? CompanyId { get; set; }
        public CompanyRoleEnum? CompanyRole { get; set; }
    }
}

File: Frontend/src/app/modules/shared/models/wep-api/domain/users/userViewModel.ts
import { CompanyRole } from "src/app/modules/shared/enums/company-role.enum";
import { UserRole } from "src/app/modules/shared/enums/user-role.enum";

export class UserViewModel {
    id!: number;
    firstName!: string;
    lastName!: string;
    role!: UserRole;
    companyId: number | undefined;
    companyRole: CompanyRole | undefined;
}
